authentication/src/main/java/io/camunda/authentication/CamundaOAuthPrincipalService.java

```diff
@@ -75,10 +79,15 @@ public OAuthContext loadOAuthContext(final Map<String, Object> claims)
       LOG.debug("No mappings found for these claims: {}", claims);
     }
 
-    final var groups =
-        groupServices.getGroupsByMemberIds(mappingIds, EntityType.MAPPING).stream()
-            .map(GroupEntity::groupId)
-            .collect(Collectors.toSet());
+    final Set<String> groups;
+    if (StringUtils.hasText(groupsClaim)) {
+      groups = new HashSet<>(oidcGroupsLoader.load(claims));
+    } else {
+      groups =
+          groupServices.getGroupsByMemberIds(mappingIds, EntityType.MAPPING).stream()
```

Comment:
❓ If you feel happy to this is also the source of a bug, we need to also include those assigned via simple mappings `USER`. If it would blur this PR too much I'm happy to look at it in a later PR of course :)

Reply:
Ah so here when we get the groups of a member we only look for those groups with mappingIds and type Mapping, but with simple mappings I may also have groups that are directly assigned to my username, in those instances the query here won't capture the `username -> USER` rows in secondary storage.

Lets leave it out of this PR though as it will bloat it too much (I think that issue also exists for the other entities too i.e. roles and tenants